windows/src/desktop/kmshell/xml/menu-frame.js

```diff
@@ -28,10 +28,17 @@
     save_state();
 
     if(!loading_state) {
-      var q = document.getElementById('content_'+c.menu_name).getElementsByTagName('span');
+      let q = null;
+      if (c.menu_name.includes("keyboardlist")) {
+        q = document.getElementById('content_'+c.menu_name).getElementsByClassName('list_item expanded');
+      }
+      // if none are expanded select the first item
+      if (q == null || q.length == 0) {
+        q = document.getElementById('content_'+c.menu_name).getElementsByClassName('list_item');
+      }
       for(var i = 0; i < q.length; i++) {
-        blah = q[i].id;
-        itemtype = blah.substr(0, 5);
+        list_array = q[i].id;
+        itemtype = list_array.substr(0, 5);
 
         if( itemtype == "list_" ) {
```

Comment:
Do we still need this check now that we get `list_item` items?

I guess we'd still need it if we could have items with class `list_item` that have an `id` not starting with `list_`.

Reply:
I think I will remove it as it is no longer really needed with the class name look up and the xsl should make sure each id starts with a list_